samples/subsys/zbus/dyn_channel/src/producer.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2022 Rodrigo Peixoto <rodrigopex@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "messages.h"
+
+#include <zephyr/zbus/zbus.h>
+
+ZBUS_CHAN_DECLARE(pkt_chan);
+
+ZBUS_SUBSCRIBER_DEFINE(producer_sub, 4);
+
+void producer_thread(void)
+{
+	struct zbus_channel *chan;
+	void *msg;
+	int i = 0;
+
+	do {
```

Comment:
Should this be a for loop? With a separate exit path for the `!zbus_sub_wait` where you could put an explicit log as well.